### PR TITLE
Circle link and  payment pointer on IPFS

### PR DESCRIPTION
**Pull request by guoliu**
as title.

### CODE REVIEW

File: src/connectors/articleService.ts [hunk 1]
```diff
@@ -108,33 +109,58 @@ export class ArticleService extends BaseService {
...
+    } as FormatterVars
+
+    // paywall info
+    if (circleId) {
+      const [
+        { name: circleName, displayName: circleDisplayName },
+      ] = await this.knex('circle')
+        .select('name', 'displayName')
+        .where({ id: circleId })
+      console.log({ name: circleName, displayName: circleDisplayName })
```
Comment (robertu7) [inline src/connectors/articleService.ts:152]: 👀

File: src/connectors/articleService.ts [hunk 2]
```diff
@@ -108,33 +109,58 @@ export class ArticleService extends BaseService {
...
       summary,
       content,
       prefix: IPFS_PREFIX,
-    })
+    } as FormatterVars
+
+    // paywall info
+    if (circleId) {
+      const [
+        { name: circleName, displayName: circleDisplayName },
```
Comment (robertu7) [inline src/connectors/articleService.ts:148]: destructuring will be failed if circleId hasn't query record?

**Review by robertu7 — APPROVED** [review]

File: src/connectors/articleService.ts (hunk 2 above)
Comment (devformatters) [reply src/connectors/articleService.ts:148]: Destructuring undefined does cause error.

**Review by devformatters — APPROVED** [review]
Looks good except lines pointed out by Robert.

**Pull request merged**